internal/dataplane/parser/translate_httproute.go

```diff
@@ -412,6 +453,47 @@ func generatePluginsFromHTTPRouteFilters(filters []gatewayv1beta1.HTTPRouteFilte
 	return kongPlugins
 }
 
+func generateRequestRedirectKongPlugin(modifier *gatewayv1beta1.HTTPRequestRedirectFilter, p string) []kong.Plugin {
+	plugins := make([]kong.Plugin, 2)
+	plugins[0] = kong.Plugin{
+		Name: kong.String("request-termination"),
+		Config: kong.Configuration{
+			"status_code": modifier.StatusCode,
+		},
+	}
+
+	var locationHeader string
+	scheme := "http"
+	port := 80
+
+	if modifier.Scheme != nil {
+		scheme = *modifier.Scheme
+	}
+	if modifier.Port != nil {
+		port = int(*modifier.Port)
+	}
+	if modifier.Path != nil && modifier.Path.Type == gatewayv1beta1.FullPathHTTPPathModifier && modifier.Path.ReplaceFullPath != nil {
+		// only ReplaceFullPath currently supported
```

Comment:
Should this 
- have its own issue to track?
- return an error if something else than `ReplaceFullPath` is used?

Reply:
I don't think so. `ReplacePath` is an experimental (and not yet used/tested) feature of the Gateway API. I implemented the `fullPathModifier` because it was straightforward. My opinion is that we can leave it as it is now, and when times are more mature, we'll think about how to address other replacements. WDYT?

Reply:
Anything that's experimental, or even optional ([this is both](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPPathModifier)) we can reasonably leave without a TODO or issue.

A comment indicating that we don't currently support some piece of optional functionality is a sufficient breadcrumb for future work if we decide we do want to support optional functionality.

We do have a UX problem where we don't support the other type, but apparently just silently ignore it as I read the code. If you set `replacePrefixMatch`, AFAICT, that config will be in place on the HTTPRoute, but the resulting Kong route won't replace anything.

I don't think we have a good standard mechanism for reporting this, and I'm not sure there's even a recommended way to alert about unsupported optional functionality from upstream.

We unfortunately can't do much here immediately, since this neither returns errors nor has access to logging. We could add it, but IMO such rejections should happen in the controller, not here in the parser. The controller is where we interact more with the parent resource, and even if we don't end up adding something to status or whatever, the controller can log just as well as parser bits can.

Reply:
Fair points from both of you 👍 In an ideal world we'd track this somewhere (an `.md` table of sorts or at least a well structure code comment) but given this is experimental I'm OK with leaving this as is.

Perhaps we could have a discussion about how to track those small features that we have implemented (similarly as we have it for high level e.g. CRDs like HTTPRoute etc.)